### PR TITLE
10 post api reviews reviewid comments

### PR DESCRIPTION
**Pull request by thatnick**

### CODE REVIEW

File: __tests__/comments.test.js [hunk 1]
```diff
@@ -50,7 +50,81 @@ describe("GET /api/reviews/:review_id/comments", () => {
...
-      msg: "sometext is not a valid review id",
+      msg: "sometext is not a valid review_id",
+    });
+  });
+});
+
+describe("POST /api/reviews/:review_id/comments", () => {
+  test("status: 201, adds the comment to the db and responds with the added comment", async () => {
+    const testStart = new Date();
+    const { body } = await request(app)
```
Comment (douglashellowell) [inline __tests__/comments.test.js:61]: you're playing with fire by using `Date` in your tests - the output may be different depending where your code is running! (who knows where the server hosting your code will be!)

I appreciate your approach and i understand that you're comparing two timestamps created one-after-the-other (hopefully 🤞) but comparing a datetime from javascript to a datetime from psql is a recipe for inconsistant tests

If you were running your tests in a pipeline (see CI-CD lecture from today) then you'd be more likely to run into trouble

File: models/errors/model-errors.js [hunk 2]
```diff
@@ -1,16 +1,20 @@
...
   return Promise.reject({
     status: 404,
-    msg: `There is no ${type} with the id ${id}`,
+    msg: `There is no ${resource} with the ${property} ${value}`,
   });
 };
 
-exports.dbError = (err, msg) => {
+dbError = (err, resource, property, value) => {
   if (err.code === "22P02") {
```
Comment (douglashellowell) [inline models/errors/model-errors.js:9]: 🚨 danger!

You are declaring global variables here! 😱

you *must* use `let` or `const` keywords

File: models/comments.model.js [hunk 3]
```diff
@@ -26,6 +26,32 @@ exports.selectCommentsByReviewId = async (review_id) => {
...
+
+    const [comment] = rows;
+    if (comment) return { comment };
+  } catch (err) {
+    if (err.detail.includes("users")) {
+      return dbError(err, "user", "username", username);
+    } else if (err.detail.includes("reviews")) {
+      return dbError(err, "review", "review_id", review_id);
+    } else {
+      return dbError(err);
```
Comment (douglashellowell) [inline models/comments.model.js:54]: i really like this, super clean

**Comment by douglashellowell** [comment]
great stuff!

ready to merge after a tweak to your tests

File: models/errors/model-errors.js (hunk 2 above)
Comment (thatnick) [reply models/errors/model-errors.js:9]: Whoops! Can't believe I didn't spot that. Perhaps need to get a linter installed..

File: __tests__/comments.test.js (hunk 1 above)
Comment (thatnick) [reply __tests__/comments.test.js:61]: That wouldn't have occurred to me but makes perfect sense. Will edit...

**Pull request merged**